site/docs-md/guides/push-notifications-firebase.md

```diff
@@ -0,0 +1,369 @@
+
+Name the project, accept the Firebase ToS and click **Create project** to continue. A Project ID should be automatically generated for you.
+
+## Android
+
+### Integrating Firebase with the Android app
+
+This section more-or-less mirrors the [setting up Firebase using the Firebase console documentation](https://firebase.google.com/docs/android/setup?authuser=0). See below for specific Capacitor-related notes.
+
+Go to the Project Overview page for your Firebase project and, under the **Grow** section, click the **Cloud Messaging** option. At the top, click on the **Android** icon.
+
+The next screen will ask you for some information about your application.
+
+- Your **Android package name** should match the **appId** from your `capacitor.config.json` file
+- We used `com.mydomain.myappname` for this Capacitor app ID, so that is what we'll use for this entry.
+- Nickname and Debug Signing Certificate are optional
+
+Then click the **Register app** button.
+
+### Download and Use the `google-services.json` file
+
+The next prompt will ask you to download a `google-services.json` file. This file contains the information your Capacitor app needs to connect to Firebase from Android.
+
+Download the `google-services.json` file to your local machine. Then move the file into your Capacitor Android project directory, specifically under `android/app/`.
+
+![Google Services JSON Location for Android](/assets/img/docs/guides/firebase-push-notifications/google-services-location-android.png)
+
+We don't need to *add* any permissions to our project because Capacitor projects automatically include a version of `firebase-messaging` in our `app/build.gradle` file:
+
+```
+implementation 'com.google.firebase:firebase-messaging:17.4.0'
+```
+
+**Note**: Make sure that the file is added to source control. You may need to modify the `.gitignore` file under `android/` to add the file.
+
+## iOS
+
+### Prerequisites
+
+iOS push notifications are significantly more complicated to set up than Android. You must have a [paid Apple Developer account](https://developer.apple.com/) *and* take care of the following items prior to being able to test push notifications with your iOS application:
+
+1. [Setup the proper Development or Production certificates & provisioning profiles](https://help.apple.com/xcode/mac/current/#/dev60b6fbbc7) for your iOS application in the Apple Developer Portal
+2. [Create an APNS certificate or key](https://developer.apple.com/documentation/usernotifications/setting_up_a_remote_notification_server/establishing_a_certificate-based_connection_to_apns) for either Development or Production in the Apple Developer Portal
+3. [Ensure Push Notification capabilities have been enabled](https://help.apple.com/xcode/mac/current/#/dev88ff319e7) in your application in Xcode
+4. Have a physical iOS device running one of the two latest iOS versions (push notification functionality does not work in the iOS simulators)
```

Comment:
Better say iOS 11 or newer, we expect to support iOS 11 when iOS 13 comes out at least for a while and that's 3 versions. Better update this to iOS 12 in the future if we drop iOS 11 support. Or link to the dependences page https://capacitor.ionicframework.com/docs/getting-started/dependencies (which I've noticed is wrong at the moment, will update it later)